src/controllers/softtakeover.h

```diff
@@ -1,56 +1,98 @@
-    void setThreshold(double threshold);
+    SoftTakeover() = default;
+    bool ignore(const ControlObject& control, double newParameter);
+    void ignoreNext() {
+        m_time = kFirstValueTime;
+    }
+    void setThreshold(double threshold) {
+        m_dThreshold = threshold;
+    }
 
-    struct TestAccess;
+    // TODO (XXX): find a better testing solution than this TestAccess
+    // front-door coupled to `mixxx::Time`.
+    struct TestAccess {
+        static constexpr mixxx::Time::duration getTimeThreshold() {
+            return kSubsequentValueOverrideTime;
+        }
+        template<class Rep = mixxx::Time::rep,
+                class Period = mixxx::Time::period>
+        static void advanceTimePastThreshold(
+                std::chrono::duration<Rep, Period> offset =
+                        std::chrono::nanoseconds(0)) {
+            mixxx::Time::addTestTime(getTimeThreshold() + offset);
+        }
+    };
 
   private:
+    using ClockT = mixxx::Time;
     // If a new value is received within this amount of time, jump to it
     // regardless. This allows quickly whipping controls to work while retaining
     // the benefits of soft-takeover for slower movements.  Setting this too
     // high will defeat the purpose of soft-takeover.
-    static const mixxx::Duration kSubsequentValueOverrideTime;
-
-    mixxx::Duration m_time;
-    double m_prevParameter;
-    double m_dThreshold;
-};
+    static constexpr ClockT::duration kSubsequentValueOverrideTime = std::chrono::milliseconds(50);
+    static constexpr ClockT::time_point kFirstValueTime = ClockT::time_point::min();
 
-struct SoftTakeover::TestAccess {
-    static mixxx::Duration getTimeThreshold() {
-        return kSubsequentValueOverrideTime;
-    }
+    ClockT::time_point m_time{kFirstValueTime};
+    double m_prevParameter{0};
+    double m_dThreshold{kDefaultTakeoverThreshold};
 };
 
 class SoftTakeoverCtrl {
   public:
-    SoftTakeoverCtrl();
-    ~SoftTakeoverCtrl();
+    SoftTakeoverCtrl() = default;
 
     // Enable soft-takeover for the given Control.
     // This does nothing on a control that already has soft-takeover enabled.
-    void enable(ControlObject* control);
+    void enable(gsl::not_null<ControlPotmeter*> pControl);
     // Disable soft-takeover for the given Control
-    void disable(ControlObject* control);
+    void disable(ControlObject* control) {
+        m_softTakeoverHash.erase(control);
+    }
     // Check to see if the new value for the Control should be ignored
-    bool ignore(ControlObject* control, double newMidiParameter);
+    bool ignore(ControlObject* pControl, double newParameter) {
+        auto coIt = m_softTakeoverHash.find(pControl);
+        if (coIt == m_softTakeoverHash.end()) {
+            return false;
+        }
+        VERIFY_OR_DEBUG_ASSERT(pControl) {
+            return false;
+        }
+        auto& [coKey, softTakeoverInstance] = *coIt;
+        return softTakeoverInstance.ignore(*pControl, newParameter);
```

Comment:
Without explicit types we need to make sure the var names are not misleading.
How about that?  
```suggestion
        auto it = m_softTakeoverHash.find(pControl);
        if (it == m_softTakeoverHash.end()) {
            return false;
        }
        VERIFY_OR_DEBUG_ASSERT(pControl) {
            return false;
        }
        auto& [key, refSoftTakeover] = *it;
        return refSoftTakeover.ignore(*pControl, newParameter);
```